src/lxc/lxccontainer.c

```diff
@@ -878,13 +976,16 @@ static bool do_lxcapi_start(struct lxc_container *c, int useinit, char * const a
 
 	/* Unshare the mount namespace if requested */
 	if (conf->monitor_unshare) {
-		if (unshare(CLONE_NEWNS)) {
+		ret = unshare(CLONE_NEWNS);
+		if (ret < 0) {
 			SYSERROR("failed to unshare mount namespace");
 			free_init_cmd(init_cmd);
 			lxc_free_handler(handler);
 			return false;
```

Comment:
If `daemonize` is true, we need  `exit(EXIT_FAILURE)` here, not the `return false`.